src/Controller/BaseController.php

```diff
@@ -6,16 +6,36 @@
 use App\Lib\Redirector;
 use App\Lib\SessionManager;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
 
 abstract class BaseController
 {
+    protected const DEFAULT_PAGE = 1;
+    protected string $defaultRoute;
+
     public function __construct(
         protected readonly SessionManager $sessionManager,
         protected readonly Redirector $redirector,
         protected readonly Renderer $renderer,
+        protected readonly RouterInterface $router
     ) {
     }
 
+    protected function getIndexRoute(?int $nb = self::DEFAULT_PAGE): string
+    {
+        return $this->router->generate($this->defaultRoute, ['nb' => $nb]);
+    }
+
+    protected function getIdFromParameters(array $parameters): ?int
+    {
+        return isset($parameters['id']);
+    }
+
+    protected function getNbFromParameters(array $parameters): ?int
+    {
+        return isset($parameters['nb']) ? (int)$parameters['nb'] : null;
```

Comment:
Il est possible de simplifier cette syntaxe :

```suggestion
        return $parameters['nb'] ?? null;
```

Reply:
Ok, merci pour le tuyau :)